test/parallel/test-http2-server-stream-session-destroy.js

```diff
@@ -24,15 +19,40 @@ server.on(
 
     // Test that ERR_HTTP2_INVALID_STREAM is thrown while calling
     // stream operations after the stream session has been destroyed
-    common.expectsError(() => stream.additionalHeaders(), errorObj);
-    common.expectsError(() => stream.priority(), errorObj);
+    const invalidStreamError = {
+      type: Error,
+      code: 'ERR_HTTP2_INVALID_STREAM',
+      message: 'The stream has been destroyed'
+    };
+    common.expectsError(() => stream.additionalHeaders(), invalidStreamError);
+    common.expectsError(() => stream.priority(), invalidStreamError);
     common.expectsError(
       () => stream.pushStream({}, common.mustNotCall()),
-      errorObj
+      invalidStreamError
     );
-    common.expectsError(() => stream.respond(), errorObj);
-    common.expectsError(() => stream.rstStream(), errorObj);
-    common.expectsError(() => stream.write('data'), errorObj);
+    common.expectsError(() => stream.respond(), invalidStreamError);
+    common.expectsError(() => stream.rstStream(), invalidStreamError);
+    common.expectsError(() => stream.write('data'), invalidStreamError);
+
+    // Test that ERR_HTTP2_INVALID_SESSION is thrown while calling
+    // session operations after the stream session has been destroyed
+    const invalidSessionError = {
+      type: Error,
+      code: 'ERR_HTTP2_INVALID_SESSION',
+      message: 'The session has been destroyed'
+    };
+    common.expectsError(() => stream.session.priority(), invalidSessionError);
+    common.expectsError(() => stream.session.settings(), invalidSessionError);
+    common.expectsError(() => stream.session.shutdown(), invalidSessionError);
+
+    // Wait for setImmediate call from destroy() to complete
+    // so that state.destroyed is set to true
+    setTimeout((session) => {
+      common.expectsError(() => session.priority(), invalidSessionError);
+      common.expectsError(() => session.settings(), invalidSessionError);
+      common.expectsError(() => session.shutdown(), invalidSessionError);
+      common.expectsError(() => session.rstStream(), invalidSessionError);
+    }, common.platformTimeout(100), stream.session);
```

Comment:
As with the other PR, please undo this change. It is the wrong approach and likely introduces unreliability rather than solving anything.

Reply:
Switched to `setImmediate` in the new commit